airbyte-cdk/python/airbyte_cdk/sources/streams/concurrent/partition_enqueuer.py

```diff
@@ -1,38 +1,52 @@
 #
 # Copyright (c) 2023 Airbyte, Inc., all rights reserved.
 #
+import time
+
+from queue import Queue
 
 from airbyte_cdk.sources.concurrent_source.partition_generation_completed_sentinel import PartitionGenerationCompletedSentinel
+from airbyte_cdk.sources.concurrent_source.thread_pool_manager import ThreadPoolManager
 from airbyte_cdk.sources.streams.concurrent.abstract_stream import AbstractStream
-from airbyte_cdk.sources.streams.concurrent.partitions.throttled_queue import ThrottledQueue
+from airbyte_cdk.sources.streams.concurrent.partitions.types import QueueItemObject
 
 
 class PartitionEnqueuer:
     """
     Generates partitions from a partition generator and puts them in a queue.
     """
 
-    def __init__(self, queue: ThrottledQueue) -> None:
+    def __init__(self, queue: Queue[QueueItemObject], thread_pool_manager: ThreadPoolManager, sleep_time_in_seconds: float = 0.1) -> None:
         """
         :param queue:  The queue to put the partitions in.
         :param throttler: The throttler to use to throttle the partition generation.
         """
         self._queue = queue
+        self._thread_pool_manager = thread_pool_manager
+        self._sleep_time_in_seconds = sleep_time_in_seconds
 
     def generate_partitions(self, stream: AbstractStream) -> None:
         """
         Generate partitions from a partition generator and put them in a queue.
         When all the partitions are added to the queue, a sentinel is added to the queue to indicate that all the partitions have been generated.
 
-        If an exception is encountered, the exception will be caught and put in the queue.
+        If an exception is encountered, it will be thrown and visible from the future it was submitted
 
         This method is meant to be called in a separate thread.
-        :param partition_generator: The partition Generator
-        :return:
         """
-        try:
-            for partition in stream.generate_partitions():
-                self._queue.put(partition)
-            self._queue.put(PartitionGenerationCompletedSentinel(stream))
-        except Exception as e:
-            self._queue.put(e)
+        for partition in stream.generate_partitions():
+            # Adding partitions to the queue generates futures. To avoid having too many futures, we throttle here. We understand that
+            # we might add more futures than the limit by throttling in the threads while it is the main thread that actual adds the
+            # future but we expect the delta between the max futures length and the actual to be small enough that it would not be an
```

Comment:
The delta was a bit more than I expected. On a run on my computer, I got a max of 10 120 futures in the list when the limit was 10 000. If you are concerned about this, I will investigate a bit more

Reply:
I'm fine with this because the limit itself doesn't really matter. we're only interested in ensuring we keep the size "small"

Reply:
I think it would be good to have some logging around this so that we can see it if we have connections where the delta explodes for some reason (e.g. different query patterns than Stripe).

Reply:
That's a good point - what if we only log if it gets, e.g., 2x bigger than we wanted?